src/sp/it/pl/main/Tmp.kt

```diff
@@ -1,14 +1,224 @@
 package sp.it.pl.main
 
+import javafx.beans.value.WritableValue
+import javafx.collections.ObservableList
 import org.reactfx.Subscription
 import sp.it.pl.gui.pane.ActionPane
+import sp.it.pl.layout.widget.controller.ClassController
 import sp.it.pl.layout.widget.controller.FXMLController
+import sp.it.pl.util.access.V
+import sp.it.pl.util.access.VarEnum
+import sp.it.pl.util.access.v
+import sp.it.pl.util.action.Action
+import sp.it.pl.util.action.IsAction
 import sp.it.pl.util.async.future.Fut
+import sp.it.pl.util.conf.Config
+import sp.it.pl.util.conf.Config.PropertyConfig
+import sp.it.pl.util.conf.Config.VarList
+import sp.it.pl.util.conf.EditMode
+import sp.it.pl.util.conf.IsConfig
+import sp.it.pl.util.conf.MainConfiguration
+import sp.it.pl.util.conf.ValueConfig
+import sp.it.pl.util.conf.computeConfigGroup
+import sp.it.pl.util.conf.obtainConfigConstraints
+import sp.it.pl.util.dev.throwIf
+import sp.it.pl.util.functional.ifNull
+import sp.it.pl.util.type.Util.getGenericPropertyType
+import sp.it.pl.util.validation.Constraint
+import sp.it.pl.util.validation.Constraint.NumberMinMax
+import java.io.File
+import java.util.function.Consumer
+import kotlin.properties.ReadOnlyProperty
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KMutableProperty
+import kotlin.reflect.KProperty
+import kotlin.reflect.full.findAnnotation
+import kotlin.reflect.jvm.isAccessible
+import kotlin.reflect.jvm.javaField
+import kotlin.reflect.jvm.javaType
 
 fun futureWrap(data: Any?): Fut<*> = data as? Fut<*> ?: Fut.fut(data)!!
 
 inline fun <reified T> ActionPane.register(vararg actions: ActionPane.ActionData<T, *>) = register(T::class.java, *actions)
 
 infix fun FXMLController.onDispose(s: () -> Unit) = d({ s() })
 
-infix fun FXMLController.initClose(s: () -> Subscription) = d(s())
\ No newline at end of file
+infix fun FXMLController.initClose(s: () -> Subscription) = d(s())
+
+fun <T: Any> ClassController.v(initialValue: T, onChange: (T) -> Unit) = VLate(initialValue, onChange)
+
+class VLate<T>(private val initialValue: T, private val onChange: (T) -> Unit) {
+    operator fun provideDelegate(ref: ClassController, property: KProperty<*>): ReadOnlyProperty<ClassController, V<T>> {
+        val v = v(initialValue) { if (ref.isInitialized) onChange(it) }
+        return object: ReadOnlyProperty<ClassController, V<T>> {
+            override fun getValue(thisRef: ClassController, property: KProperty<*>): V<T> {
+                return v
+            }
+        }
+    }
+}
+
+fun <T> Config<T>.isEditableByUser() = isEditable.isByUser && constraints.asSequence().none { it is Constraint.ReadOnlyIf && it.condition()}
+
+fun <T: Any> c(initialValue: T): ConfS<T> = ConfS(initialValue).but(Constraint.ObjectNonNull())
```

Comment:
I'm still thinking about this. The problem is that we would end up with confValueNullableList or confValueNullableReadOnly and similar stuff. I know I abuse short names, but I think this might just be the case where it's worth it. I'm open to suggestions. Note that the name must denote that value s configurable and may have nullable and potentially readOnly in the name.